pkg/metrics/metrics.go

```diff
@@ -142,3 +144,19 @@ func EnableMetrics(address string) {
 	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
 	http.ListenAndServe(address, nil)
 }
+
+// The filtered metric labels must be passed last and in the exact order of consts.KnownMetricLabelFilters.
+func FilterMetricLabels(labels ...string) []string {
+	offset := len(labels) - len(consts.KnownMetricLabelFilters)
```

Comment:
I think it would be a bit easier for developers if `FilterMetricLabels` took only values corresponding to `KnownMetricLabelFilters` as arguments instead of doing this offsetting.

Reply:
Happy to revise it in this direction, just some points for consideration -

If we were to change it to only accept and only filter the global configuration variables,
every time a developer would look to use the function,
they'd have to handle appending slices for the metric definition and incrementing the metric.

The way this is implemented, as long as your label filters are in order and come last you can just wrap your values in this function and the rest is handled (for both the metric definition and the statements incrementing the metric).

Do we still want to revise and drop the offset and move the slice operations outside of the function?
As an alternative, would it help if we expanded on the original comment instead highlight the expected usage?

Reply:
I see your point. I think this solution is good too, but `append` shouldn't be needed when updating metric, it would be something like
```
WithLabelValues("some-type", metrics.FilterMetricLabels(namespace, workload, pod, binary)...).Inc()
```
Am I missing something here?

Reply:
When I was working through the implementation, I was expecting that we could handle it that way too.
I ran into some errors; I just double-checked and had the same issues.

With this approach:
```
EventsProcessed.WithLabelValues(eventType, metrics.FilterMetricLabels(namespace, workload, pod, binary)...).Inc()
```

Golang does not like it - even wrapping parenthesis around it does not fix it.
```
# github.com/cilium/tetragon/pkg/metrics/eventmetrics
pkg/metrics/eventmetrics/eventmetrics.go:96:48: too many arguments in call to EventsProcessed.WithLabelValues
        have (string, []string)
        want (...string)
make: *** [Makefile:163: tetragon] Error 1
```

I think that's how I arrived at the `append( )` approach before spending some time revising it.
I think the current usage as a wrapping function call makes it easier to read the code and use the function,
but that is just an opinion and of course I'm open to being wrong and/or going in another direction (like `append()`)!